Sources/TuistKit/Utils/XCResultParser.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import XCResultKit
+
+final class XCResultParser {
+    private let resultFile: XCResultFile
+    private let invocationRecord: ActionsInvocationRecord
+
+    init?(url: URL) {
+        let resultFile = XCResultFile(url: url)
+        guard let invocationRecord = resultFile.getInvocationRecord() else { return nil }
+        self.resultFile = resultFile
+        self.invocationRecord = invocationRecord
+    }
```

Comment:
```suggestion
    parse(_ path: AbsolutePath) throws -> XCResultFile {
        return XCResultFile(url: url)
    }
```

We can also consider wrapping `XCResultFile` in our own model to make it easier to mock. Optional, though.